moped-editor/src/views/projects/projectView/ProjectMilestones.js

```diff
@@ -52,262 +205,220 @@ const ProjectMilestones = ({ projectId, loading, data, refetch }) => {
     },
-    {
-      title: "Related phase",
-      field: "moped_milestone",
-      editable: "never",
-      cellStyle: {
-        fontFamily: typography.fontFamily,
-        fontSize: "14px",
-      },
-      customSort: (a, b) => {
-        const aPhaseName =
-          phaseNameLookup(data)[a.moped_milestone.related_phase_id];
-        const bPhaseName =
-          phaseNameLookup(data)[b.moped_milestone.related_phase_id];
-        if (aPhaseName > bPhaseName) {
-          return 1;
-        }
-        if (aPhaseName < bPhaseName) {
-          return -1;
-        }
-        return 0;
+    [rowModesModel]
+  );
+
+  // when a user cancels editing by clicking the X in the actions
+  const handleCancelClick = (id) => () => {
+    setRowModesModel({
+      ...rowModesModel,
+      [id]: { mode: GridRowModes.View, ignoreModifications: true },
+    });
+    const editedRow = rows.find((row) => row.project_milestone_id === id);
+    if (editedRow.isNew) {
+      setRows(rows.filter((row) => row.id !== id));
+    }
+  };
+
+  // adds a blank row to the table and updates the row modes model
+  const onClickAddMilestone = () => {
+    // use a random id to keep track of row in row modes model and data grid rows
+    // before the record is added to the db
+    const id = uuidv4();
+    setRows((oldRows) => [
+      {
+        id,
+        milestone_id: null,
+        description: null,
+        date_actual: null,
+        date_estimate: null,
+        completed: false,
+        isNew: true,
+        project_milestone_id: id,
       },
-      render: (milestone) =>
-        phaseNameLookup(data)[milestone.moped_milestone.related_phase_id] ?? "",
-      width: "14%",
-    },
-    {
-      title: "Completion estimate",
-      field: "date_estimate",
-      render: (rowData) =>
-        rowData.date_estimate
-          ? format(parseISO(rowData.date_estimate), "MM/dd/yyyy")
-          : undefined,
-      editComponent: (props) => (
-        <DateFieldEditComponent
-          {...props}
-          name="date_estimate"
-          label="Completion estimate"
-        />
-      ),
-      width: "13%",
-    },
-    {
-      title: "Date completed",
-      field: "date_actual",
-      render: (rowData) =>
-        rowData.date_actual
-          ? format(parseISO(rowData.date_actual), "MM/dd/yyyy")
-          : undefined,
-      editComponent: (props) => (
-        <DateFieldEditComponent
-          {...props}
-          name="date_actual"
-          label="Date (actual)"
-        />
-      ),
-      width: "13%",
-    },
-    {
-      title: "Complete",
-      field: "completed",
-      lookup: { true: "Yes", false: "No" },
-      editComponent: (props) => (
-        <ToggleEditComponent {...props} name="completed" />
-      ),
-      width: "10%",
-    },
-  ];
+      ...oldRows,
+    ]);
+    setRowModesModel((oldModel) => ({
+      ...oldModel,
+      [id]: { mode: GridRowModes.Edit, fieldToFocus: "milestone_id" },
```

Comment:
updated to focus here when adding a new row -- I had it as fieldToFocus project_funding_id